### ISSUE TITLE
setup.py fails with invalid syntax

### ISSUE THREAD

**Issue by thomassligon** [ICB-DCM/AMICI#803]
Question: How should this be used?
AMICI/python/sdist/setup.py line 22 imports from custom_commands
AMICI/python/sdist/custom_commands.py line 16 imports from amici.setuptools
AMICI/python/sdist/amici/setuptools.py line 1: ../../amici/setuptools.py invalid syntax
AMICI/python/sdist/amici/__init__.py line 1: ../../amici/__init__.py invalid syntax

**Comment by FFroehlich** [comment]
We generally recommend installation of amici through pip: 

```
pip3 install amici
```

In theory you could also run

```
pip3 install .
```

from the base directory, but this is currently broken (https://github.com/ICB-DCM/AMICI/issues/716), so you will have to use

```
pip3 install . -e
```

**Comment by thomassligon** [comment]
Yes, that's also my preference, but it takes the AMICI sources from the Python site, and I wanted to be able to modify a source file as a way of understanding the errors, so that's why I tried installing from the cloned copy of AMICI.

**Comment by FFroehlich** [comment]
The best way to achieve this is using the script https://github.com/ICB-DCM/AMICI/blob/master/scripts/installAmiciSource.sh

**Comment by thomassligon** [comment]
Just 2 remarks for the moment:
1) AMICI/python/sdist/amici/init.py still looks very odd. I don't recognize it as valid Python, and, if it is, it would be an endless loop.
2) The scripts are all .sh, i.e. BASH shell scripts, which don't automatically run on Windows. I activated WSL (Windows Services for Linux) and installed Ubuntu on my machine, but, for the moment, this raises so many new questions that I will need time to get it running.

**Comment by thomassligon** [comment]
I tried
  pip -e . 
  and I'm getting the same Python syntax error in AMICI\python\sdist\amici\__init__.py

    ERROR: Command errored out with exit status 1:
     command: 'c:\program files (x86)\microsoft visual studio\shared\python37_64\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\data\\Tom\\Research\\AMICI\\python\\sdist\\setup.py'"'"'; __file__='"'"'C:\\data\\Tom\\Research\\AMICI\\python\\sdist\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info
         cwd: C:\data\Tom\Research\AMICI\python\sdist\
    Complete output (10 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\data\Tom\Research\AMICI\python\sdist\setup.py", line 22, in <module>
        from custom_commands import (my_install, my_build_clib, my_develop,
      File "C:\data\Tom\Research\AMICI\python\sdist\custom_commands.py", line 16, in <module>
        from amici.setuptools import generateSwigInterfaceFiles
      File "C:\data\Tom\Research\AMICI\python\sdist\amici\__init__.py", line 1
        ../../amici/__init__.py
        ^
     SyntaxError: invalid syntax
     ----------------------------------------
    ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.

Python is telling me that this is not valid Python code:
  ../../amici/__init__.py

**Comment by FFroehlich** [comment]
Sound like '..' might be invalid for path specification. I have tried to fix this in https://github.com/ICB-DCM/AMICI/pull/807/commits/08fae1dad585572e581761142d3df1d1c4ff42c4

**Comment by thomassligon** [comment]
Hi Fabian, the link in your latest post shows lines 59-84 of python/amici/__init__.py.
My problem is in AMICI\python\sdist\amici\__init__.py, which contains ONLY ONE LINE, and that refers to itself. In addition to the syntax error, it looks like everything is missing, and if the syntax were OK, the script would do nothing other than to call itself.

**Comment by FFroehlich** [comment]
`AMICI\python\sdist\amici__init__.py` should not exist. The fact that it exists suggests that the '\' between `amici` and `__init__.py` went missing somewhere, probably because some paths were not specified using `os.path.join` but manuall concatenated.

**Comment by thomassligon** [comment]
Sorry, I meant AMICI\python\sdist\amici\_init_.py. It looks like copy and paste from the original message construed backslash underline underline as something like underline alone.
Ouch, the Preview shows this wrong. Let me try again
`AMICI\python\sdist\amici\_init_.py`

**Comment by FFroehlich** [comment]
Oh, now I understand the issue. That's not a file, but a symlink, which will only work on unix systems. It is unlikely there is a quick or easy solution to this issue.

When trying to install/use amici under windows, I would generally recommend using the Windows Subsystem for Linux (WSL).

**Comment by thomassligon** [comment]
That answers my question. Doing a local installation should be useful if we ever get it working on Windows, but I don't really need it.
